src/test/scala/org/constellation/domain/transaction/TransactionChainServiceTest.scala

```diff
@@ -37,17 +39,145 @@ class TransactionChainServiceTest
+    "for dummy transaction" - {
+      "should not modify lastAcceptedTransactionRef" in {
+        val dummyTxEdge = mock[Edge[TransactionEdgeData]]
+        val dummyTx = Transaction(dummyTxEdge, LastTransactionRef.empty, isDummy = true)
+        service.acceptTransaction(dummyTx).unsafeRunSync
+
+        val result = service.lastAcceptedTransactionRef.get.unsafeRunSync
+
+        result shouldEqual Map.empty
+      }
+    }
+    "for normal transaction" - {
+      "should add transaction to lastAcceptedTransactionRef" in {
+        val src = "DAG123"
+        val dst = "DAG456"
+        val txEdge = createTransactionEdge(src, dst, LastTransactionRef.empty, 1L, Fixtures.tempKey)
+        val tx = Transaction(txEdge, LastTransactionRef.empty)
+        service.acceptTransaction(tx).unsafeRunSync
+
+        val result = service.lastAcceptedTransactionRef.get.unsafeRunSync
+
+        result shouldEqual Map(src -> LastTransactionRef(tx.hash, 1L))
+      }
+    }
   }
 
   "createAndSetLastTransaction" - {
     "should return new transaction" in {
       val src = "unknown"
+      val dst = "bb"
       val createdTx = service
-        .createAndSetLastTransaction(src, "bb", 1L, Fixtures.tempKey, false)
+        .createAndSetLastTransaction(src, dst, 1L, Fixtures.tempKey, false)
         .unsafeRunSync
 
       createdTx.src.address shouldBe src
+      createdTx.dst.address shouldBe dst
+      createdTx.lastTxRef shouldBe LastTransactionRef.empty
+      createdTx.ordinal shouldBe 1L
+      createdTx.isDummy shouldBe false
+      createdTx.amount shouldBe 100000000L
+
+      val newTx = service
+        .createAndSetLastTransaction(src, dst, 1L, Fixtures.tempKey, false)
+        .unsafeRunSync
+
+      newTx.lastTxRef shouldBe LastTransactionRef(createdTx.hash, createdTx.ordinal)
+      newTx.ordinal shouldBe 2L
+    }
+
+    "should set the last transaction properly" in {
+      val src = "unknown"
+      val createdTx = service
+        .createAndSetLastTransaction(src, "bb", 1L, Fixtures.tempKey, false)
+        .unsafeRunSync
+
+      val lastTx = service.lastTransactionRef.get.unsafeRunSync.getOrElse(src, Map.empty)
+
+      lastTx shouldEqual LastTransactionRef(createdTx.hash, createdTx.ordinal)
+    }
+  }
+
+  "applySnapshotInfo" - {
+    "should update lastAcceptedTransactionRef" in {
+      val lastAcceptedTransactionRef = Map("DAG123" -> LastTransactionRef("abc", 1), "DAG456"-> LastTransactionRef("def", 1))
+      val snapshotInfo = mock[SnapshotInfo]
+      snapshotInfo.lastAcceptedTransactionRef shouldReturn lastAcceptedTransactionRef
+      service.applySnapshotInfo(snapshotInfo).unsafeRunSync
+
+      val result = service.lastAcceptedTransactionRef.get.unsafeRunSync
+      result shouldEqual lastAcceptedTransactionRef
+    }
+  }
+
+  "checkDummyTransaction" - {
+    val src = "DAG123"
+    val dst = "DAG456"
+    "should create a dummy transaction with an empty transaction ref even if isDummy is set to false" in {
```

Comment:
if `check` should `create` we have something wrong with this function. can you track where it's used and if it's needed?

Reply:
It's used in createTransaction next to the transactionChainService.createAndSetLastTransaction.
The case when we are processing a dummy transaction.